mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionMetadataTypeUtils.java

```diff
@@ -68,6 +70,15 @@ public static boolean isFinal(MetadataType metadataType) {
     }
   }
 
+  public static boolean isMap(MetadataType metadataType) {
+    boolean implementsMap = metadataType.getAnnotation(ClassInformationAnnotation.class)
```

Comment:
this is wrong. If the metadataType has a typeId and JAVA format, then what you should do is `Map.class.isAssignableFrom(theType)`. If typeId is not present, only then go to find the ClassInformation